browser/src/control/Control.Tabs.js

```diff
@@ -283,17 +281,27 @@ L.Control.Tabs = L.Control.extend({
 		this._map.sendUnoCommand('.uno:Move?Copy:bool=false&UseCurrentDocument:bool=true&Index=' + newIndex);
 	},
 
+	_moveOrCopySheet: function () {
+		var contextMenuTab = this._tabForContextMenu;
+		this._map.sendUnoCommand('.uno:Move?FromContextMenu:bool=true&MoveOrCopySheetDialog:bool=true&ContextMenuIndex=' + contextMenuTab);
+	},
+
+	_moveSheetLR: function (contextMenuTab, contextMenuTargetTab) {
+		if (contextMenuTab !== undefined && contextMenuTab >= 0)
+			this._map.sendUnoCommand('.uno:Move?Copy:bool=false&UseCurrentDocument:bool=true&FromContextMenu:bool=true&ContextMenuIndex=' + contextMenuTab + '&ContextMenuTargetIndex=' + contextMenuTargetTab);
+	},
+
 	_moveSheetLeft: function () {
-		var targetIndex = this._map._docLayer._partNames.indexOf(this._map._docLayer._partNames[this._map._docLayer._selectedPart]);
 		//core handles sheet with 1 base indexing
 		// 0 index means last sheet
-		if (targetIndex <= 0) return;
-		this._moveSheet(targetIndex);
+		var contextMenuTab = this._tabForContextMenu;
+		if (contextMenuTab <= 0) return;
+		this._moveSheetLR(contextMenuTab, contextMenuTab);
 	},
 
 	_moveSheetRight: function () {
-		var targetIndex = this._map._docLayer._partNames.indexOf(this._map._docLayer._partNames[this._map._docLayer._selectedPart]) + 3;
-		this._moveSheet(targetIndex);
+		var contextMenuTab = this._tabForContextMenu;
+		this._moveSheetLR(contextMenuTab, contextMenuTab + 3);
```

Comment:
Why +3 ?

Reply:
The old code, which is `var targetIndex = this._map._docLayer._partNames.indexOf(this._map._docLayer._partNames[this._map._docLayer._selectedPart]) + 3;` also use `+3`, and when I was trying to understand why there is `+3`; I figured out that:
- for moveRight, core side acts like 1-based index. so we need to add `+1`
- since we moveRight we need to add `+1`
- and at the core side, there is a `-1` for this value, so we add `+1` again.

I will send a new patchset that will remove the `-1` in the core, so we will only use `+2` for the moveRight.

Reply:
So rather adjust `ContextMenuIndex` to be also compatible with `Index` param in the core...

Reply:
OK. I will do so.